apps/prairielearn/src/pages/administratorQueries/administratorQueries.ts

```diff
@@ -16,15 +20,19 @@ router.get(
   asyncHandler(async (req, res, _next) => {
     const fileList = await fsPromises.readdir(queriesDir);
     const jsonList = fileList.filter((f) => /\.json$/.test(f));
-    const queries = await Promise.all(
+    const queries: AdministratorQuery[] = await Promise.all(
       jsonList.map(async (f) => {
         const contents = AdministratorQueryJsonSchema.parse(
           await jsonLoad.readJSON(path.join(queriesDir, f)),
         );
+        const filePrefix = f.replace(/\.json$/, '');
         return {
           ...contents,
-          sqlFilename: f.replace(/\.json$/, '.sql'),
-          link: f.replace(/\.json$/, ''),
+          type: await import(path.join(queriesDir, `${filePrefix}.js`)).then(
+            () => 'Module',
+            () => 'SQL',
+          ),
```

Comment:
This may be overkill just to show the query type. If knowing the type in the queries list is not important I'm happy to remove it. We could also use a heuristic instead (e.g., check if there is a file with TS or JS extension).